### PR TITLE
DAOS-1967 control: specify targets and related config parameters per-server

### PR DESCRIPTION
**Pull request by tanabarr**
Move targets and related configuration file parameters from global
section into per-server section so they can be set independently for
different daos_io_servers on the same storage node.

Add unmarshal yaml method for "server" type (list of server type
objects embedded in config type parsed by control plane) to apply
default values (NrXsHelpers), amend tests appropriately. 

Attempt to simplify implementation of config file parsing.

### CODE REVIEW

**Comment by tanabarr** [comment]
gatekeeper please use first comment in PR as commit message

File: src/control/server/commands.go [hunk 1]
```diff
@@ -33,20 +33,20 @@ import (
...
-	Map        *string `short:"y" long:"map" description:"[Temporary] System map file"`
-	Rank       *rank   `short:"r" long:"rank" description:"[Temporary] Self rank"`
-	SocketDir  string  `short:"d" long:"socket_dir" description:"Location for all daos_server & daos_io_server sockets"`
-	Storage    StorCmd `command:"storage" alias:"st" description:"Perform tasks related to locally-attached storage"`
+	Port        uint16  `short:"p" long:"port" description:"Port for the gRPC management interfect to listen on"`
+	MountPath   string  `short:"s" long:"storage" description:"Storage path"`
+	ConfigPath  string  `short:"o" long:"config_path" description:"Server config file path"`
+	Modules     *string `short:"m" long:"modules" description:"List of server modules to load"`
+	Cores       uint16  `short:"c" long:"cores" default:"0" description:"option deprecated, please use targets instead"`
+	Targets     uint16  `short:"t" long:"targets" default:"0" description:"number of targets to use (default use all cores)"`
```
Comment (jbrosenz) [inline src/control/server/commands.go]: I'm not up to speed on the rationale of changing from 'cores' to 'targets'.  However, the new help text mixes 'targets' and 'cores' and it's not clear why.  It says ' ... (default use all cores) '.  It seems like it should say ' ... (default use all targets)'.

Comment (tanabarr) [reply src/control/server/commands.go]: targets run on cores, this was xuezhao's wording and seems okay to me. it's basically saying that by default you have as many targets as you have cores.

File: src/control/server/config.go [hunk 2]
```diff
@@ -242,42 +265,40 @@ func (c *configuration) cmdlineOverride(opts *cliOptions) {
...
-			c.Servers[i].Cpus, _ = setNumCores(opts.Targets)
-		} else if c.Targets > 0 {
-			c.Servers[i].Cpus, _ = setNumCores(c.Targets)
+			srv.Targets, _ = setNumCores(int(opts.Targets))
 		}
-		if opts.Rank != nil {
-			// override first per-server config (doesn't make sense
-			// to reply to more than one server)
-			c.Servers[0].Rank = opts.Rank
+		if *opts.NrXsHelpers != 2 {
```
Comment (kjacque) [inline src/control/server/config.go]: Not sure I totally get this check - This is the default, right? Any reason we wouldn't want to set srv.NrXsHelpers to this?

File: src/control/server/config_test.go [hunk 3]
```diff
@@ -372,28 +390,31 @@ func TestSetNumCores(t *testing.T) {
...
+
 		AssertEqual(t, num, tt.num, "failed to convert to expected number")
 	}
 }
 
 // TestCmdlineOverride verified that cliOpts take precedence over existing
 // configs resulting in overrides appearing in ioparams
 func TestCmdlineOverride(t *testing.T) {
-	r := rank(9)
+	//r := rank(9)
```
Comment (kjacque) [inline src/control/server/config_test.go]: Did you mean to leave these lines commented out?

File: src/control/server/config_test.go [hunk 4]
```diff
@@ -402,239 +423,280 @@ func TestCmdlineOverride(t *testing.T) {
...
-					"-g", "daos",
-					"-s", "/mnt/daos/2",
-					"-x", "0",
-					"-r", "1",
-					"-d", "./.daos/daos_server",
-				},
-			},
-			desc: "Targets config overrides Cores cli",
-		},
+		//		{
```
Comment (kjacque) [inline src/control/server/config_test.go]: Some more that's commented out...

File: src/control/server/config.go (hunk 2 above)
Comment (tanabarr) [reply src/control/server/config.go]: because that would override a setting from the config file

File: src/control/server/config_test.go (hunk 3 above)
Comment (tanabarr) [reply src/control/server/config_test.go]: no, thanks for spotting

File: src/control/server/commands.go (hunk 1 above)
Comment (liuxuezhao) [reply src/control/server/commands.go]: -t configures the number of vos target, each vos tgt can with multiple XS thread (-x number + 1).
the original -c is not really reasonable as it is not number of cores.

**Pull request merged**